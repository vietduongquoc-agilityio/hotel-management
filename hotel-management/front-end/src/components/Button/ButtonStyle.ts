import { ButtonProps } from "@chakra-ui/react";

export type ButtonType =
  | "first"
  | "paginationButton"
  | "cancelButton"
  | "deleteButton"
  | "nextButton";

export const getStyleButton = (type: ButtonType): ButtonProps => {
  switch (type) {
    case "first":
      return first;
    case "paginationButton":
      return paginationButton;
    case "cancelButton":
      return cancelButton;
    case "deleteButton":
      return deleteButton;
    default:
      return nextButton;
  }
};

const first: ButtonProps = {
  fontSize: "md",
  variant: "solid",
  fontWeight: "bold",
};

const nextButton: ButtonProps = {
  fontSize: "md",
  variant: "solid",
  fontWeight: "bold",
  color: "grey.500",
  _hover: {
    bg: "blue.100",
    border: "1px solid #1570ef",
    color: "blue.500",
  },
  width: "90px",
  height: "36px",
  bg: "white.200",
  borderRadius: "8px",
  border: "1px solid #667085",
};

const paginationButton: ButtonProps = {
  border: "1px solid #ffffff",
  height: "40px",
  size: "sm",
  width: "40px",
  color: "grey.400",
  _hover: { color: "blue.500", bg: "blue.100" },
  _active: {
    color: "blue.500",
  },
};

const cancelButton: ButtonProps = {
  fontSize: "md",
  variant: "solid",
  fontWeight: "bold",
  bg: "white.200",
  border: "1px solid #858d9d",
  color: "grey.500",
  _hover: {
    bg: "error.400",
    color: "white.200",
    border: "none",
  },
  marginRight: "20px",
};

const deleteButton: ButtonProps = {
  fontSize: "md",
  variant: "solid",
  fontWeight: "bold",
  bg: "error.400",
  border: "none",
  color: "white.200",
  _hover: {
    bg: "red",
  },
};
