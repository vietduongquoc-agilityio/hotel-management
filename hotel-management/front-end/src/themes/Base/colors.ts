export const colors = {
  blue: {
    100: "#d4e5fa",
    400: "#589af7",
    500: "#1570ef",
    600: "#1366d9",
    800: "#0c3e83",
  },
  grey: {
    50: "#f0f1f3",
    100: "#f0d3d9",
    300: "#989fad",
    500: "#667085",
    400: "#858d9d",
    600: "#5d6679",
    700: "#48505e",
    800: "#383e49",
    900: "#2b2f38",
  },
  success: {
    50: "#e7f8f0",
    400: "#41c588",
  },
  error: {
    50: "#feeceb",
    400: "#f36960",
  },
  warning: {
    50: "#fef4e6",
    400: "#f9a63a",
  },
  white: {
    200: "#ffffff",
  },
};
