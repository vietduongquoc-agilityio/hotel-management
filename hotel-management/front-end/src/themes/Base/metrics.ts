import { ThemeOverride } from "@chakra-ui/react";
export const borderRadius: ThemeOverride["borderRadius"] = {
  none: "0",
  sm: "4px",
  md: "8px",
  lg: "12px",
  xl: "16px",
};

export const sizes: ThemeOverride["sizes"] = {
  sm: "320px",
  md: "768px",
  lg: "1024px",
  xl: "1280px",
  "2xl": "1536px",
};
