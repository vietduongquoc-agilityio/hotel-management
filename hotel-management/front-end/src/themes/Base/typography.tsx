import { ThemeOverride } from "@chakra-ui/react";
import { Global } from "@emotion/react";

export const fontSizes: ThemeOverride["fontSizes"] = {
  xs: "10px",
  sm: "12px",
  md: "14px",
  lg: "18px",
  xl: "32px",
  "2xl": "24px",
  "3xl": "28px",
  "4xl": "36px",
};

export const Fonts = () => (
  <Global
    styles={`
      
      @font-face {
        font-family: 'Inter Regular';
        font-style: normal;
        font-weight: 400;
        font-display: swap;
        src: url('../../assets/fonts/inter-regular.ttf') format('truetype');
      }
      
      @font-face {
        font-family: 'Inter Medium';
        font-style: normal;
        font-weight: 500;
        font-display: swap;
        src: url('../../assets/fonts/inter-medium.ttf') format('truetype');
      }

       @font-face {
        font-family: 'Inter Semibold';
        font-style: normal;
        font-weight: 600;
        font-display: swap;
        src: url('../../assets/fonts/inter-semibold.ttf') format('truetype');
      }
      `}
  />
);
