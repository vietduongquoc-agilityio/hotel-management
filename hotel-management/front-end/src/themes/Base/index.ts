import { colors } from "./colors";
import { borderRadius, sizes } from "./metrics";
import { Fonts, fontSizes } from "./typography";

export const bases = {
  colors,
  fontSizes,
  Fonts,
  borderRadius,
  sizes,
};
