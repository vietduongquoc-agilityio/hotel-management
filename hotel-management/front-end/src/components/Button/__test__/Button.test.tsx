import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { ChakraProvider } from "@chakra-ui/react";
import Button from "../";
import { themeColor } from "@/themes/Base/colors";
import React from "react";

// Wrapper component for tests to include ChakraProvider with the custom theme
const Wrapper = ({ children }: { children: React.ReactNode }) => (
  <ChakraProvider theme={themeColor}>{children}</ChakraProvider>
);

describe("Button Component", () => {
  it("renders the button with the correct text", () => {
    const { container } = render(
      <Button buttonType="primary" text="Click Me" onClick={() => {}} />,
      { wrapper: Wrapper }
    );
    expect(container).toMatchSnapshot();
  });

  it("applies the correct styles for pagination button", () => {
    render(<Button buttonType="pagination" text="1" onClick={() => {}} />, {
      wrapper: Wrapper,
    });
    const button = screen.getByText(/1/i);
    expect(button).toHaveStyle("border: 1px solid #ffffff");
    expect(button).toHaveStyle("height: 40px");
    expect(button).toHaveStyle("width: 40px");
  });

  it("applies additional styles passed through ButtonProps", () => {
    render(
      <Button
        buttonType="primary"
        text="Styled Button"
        onClick={() => {}}
        size="lg"
        variant="solid"
      />,
      { wrapper: Wrapper }
    );
    const button = screen.getByText(/Styled Button/i);
    expect(button).toHaveClass("chakra-button");
  });
});
