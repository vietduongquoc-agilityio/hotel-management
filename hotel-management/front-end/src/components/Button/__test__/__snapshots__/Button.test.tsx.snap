// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders the button with the correct text 1`] = `
<div>
  <button
    class="chakra-button css-rbc43c"
    type="button"
  >
    Click Me
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`Button Component renders the button with the correct text 2`] = `
<div>
  <button
    class="chakra-button css-i8iae6"
    type="button"
  >
    Click Me
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`Button Component renders the button with the correct text 3`] = `
<div>
  <button
    class="chakra-button css-kvt2wa"
    type="button"
  >
    Click Me
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
